mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -609,26 +611,23 @@ private[clustering] object OnlineLDAOptimizer {
    *         statistics for updating lambda and `ids` - list of termCounts vector indices.
    */
   private[clustering] def variationalTopicInference(
-      termCounts: Vector,
+      indices: List[Int],
```

Comment:
Yes I think this kind of approach is what I had in mind, but keep code that converts new .ml vectors to indices/values in the .ml package.